Added: COOP (10-22) Hasty Ambush

TheLocalPub opened the pull request:
Added a new coop mission.

Til-Weimann reviewed (changes requested):
- US minor victory will also run when there's a major victory, because the additional condition of the major one is not included - you would add another "not infdead" here.
- You can just use conjunctions instead of matching the count of all literals.
- The OpforAlive exfil trigger is not going to work, since it compares the local to the global amount of ALIVE PLAYERS (it takes neither AI nor casualties into account). Either change it to a faction trigger (stop after first change enabled) where the US has to e. g. make up at least 50%, or better stick to BTRs as the condition until there's something in the framework like arc suggested (global faction AI casualty flag).
- OpforExtract trigger is currently a presence trigger, you could also change this to a faction capture trigger if there were still any performance issues.
- If the move waypoint priority is this high (1500), AI will just blast through there with only the gunner returning fire. Infantry may not fight at all and just continue running, even if forced to dismount.
- Blufor Extract seems a bit small, perhaps also disable SAFC here so it cannot accidentally be triggered in the beginning and stay that way.

TheLocalPub commented:
How does waypoint priority work?
I assumed it was "X waypoint is more important then Y waypoint". I did turn it down to like 500 once I'm sure and it made no difference on the speed vehicles moved?

joewhite94 commented:
> How does waypoint priority work? I assumed it was "X waypoint is more important then Y waypoint". I did turn it down to like 500 once I'm sure and it made no difference on the speed vehicles moved?

Not vehicle speed, Til's saying at a certain point the waypoint will be seen as higher priority than actual self preservation or combat so the AI will make no effort to fight back when ambushed and just keep on moving

TheLocalPub commented:
This explains why the only difference between WP_Move and WP_ForcedMoved is the priority. 
Wp_move is set to 0.
I just lowered it to 500. Is this bad still?

TheLocalPub commented:
Okay.

> * US minor victory will also run when there's a major victory, because the additional condition of the major one is not included - you would add another "not infdead" here.
> * You can just use conjunctions instead of matching the count of all literals.
> * The OpforAlive exfil trigger is not going to work, since it compares the local to the global amount of ALIVE PLAYERS (it takes neither AI nor casualties into account). Either change it to a faction trigger (stop after first change enabled) where the US has to e. g. make up at least 50%, or better stick to BTRs as the condition until there's something in the framework like arc suggested (global faction AI casualty flag).
> * OpforExtract trigger is currently a presence trigger, you could also change this to a faction capture trigger if there were still any performance issues.
> * If the move waypoint priority is this high (1500), AI will just blast through there with only the gunner returning fire. Infantry may not fight at all and just continue running, even if forced to dismount.
> * Blufor Extract seems a bit small, perhaps also disable SAFC here so it cannot accidentally be triggered in the beginning and stay that way.

- So I believe i've sorted out the first point with two flags that turn on and off.
- I've kept the Opfor extract for the BTRs as it was fine with no performance issue.
- Adjusted waypoint priority to be lower to the 250 mark.
- Adjusted blufor extract size and toggled SAFC
- Changed end conditions to be 2x BTRs destroyed to be a minor win or 4x BTRs destroyed to be a major, awaiting an AI alive and/or dead counter.

Til-Weimann commented:
> So I believe i've sorted out the first point with two flags that turn on and off.

It's a bit more complicated than it needs to be, but this should work.

> Adjusted waypoint priority to be lower to the 250 mark.

That's still way too high if you want AI to dismount and fight after being ambushed. Try something like 30.

> Adjusted blufor extract size and toggled SAFC

You didn't actually disable SAFC, though it's not a significant issue.

> Changed end conditions to be 2x BTRs destroyed to be a minor win or 4x BTRs destroyed to be a major, awaiting an AI alive and/or dead counter.

Looks good, but briefing and end messages still mention 25% infantry.

Also, I highly recommend increasing the waypoint delay on the enemy vehicles by a couple of minutes.
Right now, the convoy starts moving after less than 30 seconds. Players are no closer to good ambush sites, and will need more time to get going.

TheLocalPub commented:
Lol
I'm glad you reminded me about the way point times haha.
That was for testing!

TheLocalPub commented:
> > So I believe i've sorted out the first point with two flags that turn on and off.
> 
> It's a bit more complicated than it needs to be, but this should work.
> 
> > Adjusted waypoint priority to be lower to the 250 mark.
> 
> That's still way too high if you want AI to dismount and fight after being ambushed. Try something like 30.
> 
> > Adjusted blufor extract size and toggled SAFC
> 
> You didn't actually disable SAFC, though it's not a significant issue.
> 
> > Changed end conditions to be 2x BTRs destroyed to be a minor win or 4x BTRs destroyed to be a major, awaiting an AI alive and/or dead counter.
> 
> Looks good, but briefing and end messages still mention 25% infantry.
> 
> Also, I highly recommend increasing the waypoint delay on the enemy vehicles by a couple of minutes. Right now, the convoy starts moving after less than 30 seconds. Players are no closer to good ambush sites, and will need more time to get going.

Okay so the reason you didn't see all my previous changes is because I didn't upload all the layers. 
I've now uploaded the layers with the fixes. The problems you mentioned about the briefing and SAFC were changed but just not uploaded.

I fixed the timing with the convoy to be 7mins as intended.

Til-Weimann commented:
> Okay so the reason you didn't see all my previous changes is because I didn't upload all the layers.
I've now uploaded the layers with the fixes. The problems you mentioned about the briefing and SAFC were changed but just not uploaded.

I can see that you fixed the convoy timing, but the changes to briefing, messages and exfil trigger are still not included.

TheLocalPub commented:
Im a little confused. I've edited all the files... 
![sddasd](https://github.com/user-attachments/assets/be4c97ee-1a61-45a6-ae3a-bed84002aec5)
![dffdfsfdf](https://github.com/user-attachments/assets/c5bfe6f8-619f-4de7-b2ff-cab7bb9b092d)

TheLocalPub commented:
Let me delete the layer folder and world file again I guess and upload with this version??? Which I believe is literally the same.

TheLocalPub commented:
Ok so as you can see.
I've deleted the world file and the whole layer directory and then reuploaded this after opening work bench, double checking, and saving.
This SHOULD 100% be everything, it should have been 100% last time, idk what's happened....

Til-Weimann reviewed (approved):
Good now

TheLocalPub commented:
HECKING YES GAMERS!
